### DISCUSSION TITLE
Typewriter text effect on a paragraph

### DISCUSSION THREAD

**Discussion by hellocatfood** [friction2d/friction#235] [Q&A]
I've created a typewriter text effect by using the diminish effect to control the opacity of the letters. It works well on a single line of text.

https://github.com/user-attachments/assets/bc965cc5-82fa-4b66-8237-80e9b87871ee

However, if I have a paragraph of text it doesn't behave as I expected.

https://github.com/user-attachments/assets/0060bfc3-dd4a-4a1e-8dfc-5892aff77445

Is there currently a way to do this effect on a paragraph of text or do I have to create a new text object for each line of text?

Attached is my friction project

[typewritertext.zip](https://github.com/user-attachments/files/16825551/typewritertext.zip)

**Comment by rodlie** [comment]
Text effects are currently limited to one line.

You could just use masking with the blend modes:

https://github.com/user-attachments/assets/558d1a28-5d31-4155-9ee7-fd9f09c4b77b

**Reply by rodlie** [reply]
https://github.com/user-attachments/assets/532b2fc4-2f9f-4feb-8627-10d818a50650

**Comment by hellocatfood** [comment]
Thanks for the suggestion. The reason I avoid using a mask for the typewriter text effect is because, depending on the speed of the mask, on some frames you can see partial letters which you wouldn't get on an actual typewriter (opacity/visibility is more like a binary in that sense).

**Comment by rodlie** [comment]
You could also animate the actual text input:


https://github.com/user-attachments/assets/fa542021-db02-4b5f-97e7-26e133b4a724

*Note that the font widget is not properly in sync when you scroll the timeline, this will be [fixed](https://github.com/friction2d/friction/issues/157).*
